src/screening/ScreeningForm.jsx

```diff
@@ -16,6 +16,14 @@ class ScreeningForm extends React.Component {
     // find any matching encounter in the active visit
     // TODO what if there are multiple encounters of the same type?  this currently just shifts in the "first"
     if (props.patient && props.patient.visit && props.patient.visit.encounters) {
+
+      // Sorts the encounters by encounterDatetime in Desc order
+      props.patient.visit.encounters.sort((a,b) => {
+        a = new Date(a.encounterDatetime);
+        b = new Date(b.encounterDatetime);
+        return a>b ? -1 : a<b ? 1 : 0;
+      });
```

Comment:
@jomadoye will this sort the patient.visit.encounters (which comes from the Redux store) "in place"?  If so, I don't think we want to do that, but rather make a local copy of the encounters list and sort that?

Reply:
@mogoodrich for this, do you mean I should do something like 
```
let encounters = props.patient.visit.encounters;
let sortedEncounters = encounters.sort((a,b) => {
        a = new Date(a.encounterDatetime);
        b = new Date(b.encounterDatetime);
        return a>b ? -1 : a<b ? 1 : 0;
      });
```

Reply:
Doing  "let encounters = props.patient.visit.encounters" will pass the variable the array by reference, and so will mutate the original array in the store... we should make a copy of the array before sorting and mutating it.

Making a shallow copy  of the array should be fine (though it occurs to me now that I should make sure that EncounterForm doesn't modify the encounter that is passed on to it, I will check that out).

See:

https://stackoverflow.com/questions/9592740/how-can-you-sort-an-array-without-mutating-the-original-array

https://redux.js.org/faq/immutabledata#what-approaches-are-there-for-handling-data-immutability-do-i-have-to-use-immutable-js

https://redux.js.org/recipes/structuringreducers/immutableupdatepatterns